apps/topics-mine/src/python/topics_mine/posts/pushshift_twitter.py

```diff
@@ -0,0 +1,291 @@
+        referenced_tweet_ids = list(quote_tweets) + list(retweets)
+
+        referenced_tweets_query_dict = self._build_id_query(referenced_tweet_ids)
+        referenced_tweets = self._fetch_tweets(referenced_tweets_query_dict)
+
+        for tweet in referenced_tweets:
+            if tweet['id'] in quote_tweets:
+                for quote_tweet in quote_tweets[tweet['id']]:
+                    quote_tweet['quoted_status'] = tweet
+            elif tweet['id'] in retweets:
+                for retweet in retweets[tweet['id']]:
+                    retweet['retweeted_status'] = tweet
+
+    def _fetch_tweets(self, query_dict: dict) -> dict:
+        """Request paginated tweets from the Pushshift API."""
+
+        tweets, scroll_id = self._fetch_tweets_page(query_dict)
+        while scroll_id:
+            scroll_dict = {'scroll': PS_TWITTER_SCROLL_TIMEOUT, 'scroll_id': scroll_id}
+            page, scroll_id = self._fetch_tweets_page(scroll_dict)
+            tweets += page
+
+        return tweets
+
+    def _fetch_tweets_page(self, query_dict: dict) -> dict:
+        """Request a single page of tweets from the Pushshift API."""
+
+        url = PS_TWITTER_URL if 'scroll_id' not in query_dict else PS_TWITTER_SCROLL_URL
+        request = Request(method='GET', url=url)
+        request.set_content_type('application/json; charset=UTF-8')
+        request.set_content(encode_json(query_dict))
+
+        ua = _get_user_agent()
+        response = ua.request(request)
+        response_str = response.decoded_content()
+
+        if not response.is_success():
+            error_msg = 'error fetching posts: %s' % response_str
+            raise McPostsPushshiftTwitterDataException(error_msg)
+
+        response_dict = dict(decode_json(response_str))
+        if 'hits' not in response_dict:
+            error_msg = 'error parsing response: %s' % response_str
+            raise McPostsPushshiftTwitterDataException(error_msg)
+        
+        shards = response_dict['_shards']
+        if shards['total'] != shards['successful'] or shards['failed'] > 0:
+            log.warning('total shards: {}, successful shards: {}, failed shards: {}'.format(
+                shards['total'], shards['successful'], shards['failed']))
+
+        page = [hit['_source'] for hit in response_dict['hits']['hits']]
+        scroll_id = None if len(page) < PS_TWITTER_PAGE_SIZE else response_dict['_scroll_id']
+        return page, scroll_id
+
+    def _parse_tweets(self, tweets: list) -> list:
+        """Parse the tweets returned from the Pushshift API."""
+
+        posts = []
+        for tweet in tweets:
+            log.debug('tweet: %d' % tweet['id'])
+            publish_date = dateutil.parser.parse(tweet['created_at']).isoformat()
+            posts.append({
+                'post_id': tweet['id_str'],
+                'data': tweet,
+                'content': tweet['text'],
+                'publish_date': publish_date,
```

Comment:
Could you verify that `publish_date` gets converted to `America/New_York` timezone at some point, in the caller perhaps? Unfortunately `topic_posts.publish_date` is a `TIMESTAMP [WITHOUT A TIME ZONE]` in PostgreSQL, so we have to normalize all the dates that come in to EST / EEST (or maybe it's UTC for `topic_posts` specifically? Either way, it has to point to the same moment of time somehow), and I don't know whether it's the caller that does that, or are you supposed to do it in the post ingest module such as this one.